src/org/mozartoz/truffle/translator/Translator.java

```diff
@@ -272,7 +277,9 @@ OzNode translate(StatOrExpr node) {
 			}
 			return SequenceNode.sequence(decls.toArray(new OzNode[decls.size()]), translate(local.body()));
 		} else if (node instanceof CallCommon) {
-			return translateCall((CallCommon) node);
+			return new TailCallCatcherNode(translateCall((CallCommon) node));
```

Comment:
Actually, move it to something like CallNode.create() and review usages of CallNodeGen.create, as they should all be wrapped with a catcher.